examples/factory/src/lib.rs

```diff
@@ -0,0 +1,423 @@
+//! the constructor, removing the separate `initialize` operation entirely. In
+//! the example presented here, the factory initializes each product with a
+//! unique index, and maintains a map of the products by this index. The
+//! `produce` and `initialize` operations are used to establish this
+//! relationship.
+//!
+//! In particular, if the factory contract does not need to update its state
+//! after initializing the product, then it serves no purpose. It would be
+//! sufficient to construct the product directly in the same way as the factory
+//! would have.
+//!
+//! ## Register the instance with a contract registry
+//! In the present example, the factory contract serves as a registry of its
+//! products. The factory pattern enables this, as the factory is responsible
+//! for initializing each product instance. However, the factory pattern is not
+//! in itself necessary for a contract registry. Instead, instances could be
+//! constructed directly (as in the previous alternative) and subsequently
+//! registered with a registry contract, which maintains an index of the
+//! instances registered with it. The registration process could also provide
+//! additional initialization information for the contract being registered
+//! (such as the index it is assigned by the registry).
+//!
+//! In the factory pattern, the factory authenticates the uninitialized product
+//! instances (checking that they are instances of a particular determined smart
+//! contract). A registry could perform similar checks. Alternatively, updating
+//! the registry could require authorization by a trusted party that eliminates
+//! the need for such checks. This may be useful if the registry is expected to
+//! handle instances of multiple different smart contracts.
+//!
+//! ## A monolithic contract
+//! Instead of having separate contract instances for the factory and each
+//! product, all of these could be combined in a single monolithic contract. In
+//! this model, each product is assigned a virtual address, and the contract
+//! maintains a mapping from virtual addresses to the state of each product.
+//!
+//! For instance, a CIS2 contract can manage multiple NFTs that each have
+//! distinct token IDs. Rather than each NFT being in its own contract (that may
+//! be created by a factory contract), they are simply handled as part of the
+//! state of the overall CIS2 contract. Here, the token ID acts as a virtual
+//! address.
+//!
+//! The main disadvantage of this approach is that the isolation between the
+//! states of each product must be enforced by the contract itself. If the state
+//! became corrupted (due to a bug in the contract) then it could affect any or
+//! all of the encapsulated products. With the factory pattern, the runtime
+//! system of the chain enforces isolation. As always, the balance of risks
+//! should be considered when choosing the approach for any application.
+
+#![cfg_attr(not(feature = "std"), no_std)]
```

Comment:
Is it checked in CI that no_std actually compiles?

Reply:
I believe it does now.